### PR TITLE
P20-1165: Implement Global Edition routing

### PR DESCRIPTION
**Pull request by artem-vavilov**
Implemented routing middleware to allow for prefixed Global Edition URLs

## Links
- JIRA task: [P20-1165](https://codedotorg.atlassian.net/browse/P20-1165)

### CODE REVIEW

File: apps/src/sites/studio/pages/teacher_dashboard/show.js [hunk 1]
```diff
@@ -83,7 +83,10 @@ $(document).ready(function () {
 
   // When removing v1TeacherDashboard after v2 launch, remove `selectedSection` from api response.
   const getV1TeacherDashboard = () => {
-    const baseUrl = `/teacher_dashboard/sections/${section.id}`;
+    const baseUrl = window.location.pathname.replace(
+      new RegExp(`(/teacher_dashboard/sections/${section.id}).*`),
+      '$1'
+    );
```
Comment (wilkie) [inline apps/src/sites/studio/pages/teacher_dashboard/show.js:92]: What does this do? Perhaps we need a comment to describe the purpose of this.

Comment (artem-vavilov) [reply apps/src/sites/studio/pages/teacher_dashboard/show.js:92]: React Router refuses to render a component if the current location path does not start with the specified `basename` (baseUrl). This problem occurs when the Global Editing prefix is added, changing the location path from `/teacher_dashboard/sections/:section_id` to `/global/fa/teacher_dashboard/sections/:section_id`:

Comment (artem-vavilov) [reply apps/src/sites/studio/pages/teacher_dashboard/show.js:89]: To resolve this issue, I strip everything from the current location path after `/teacher_dashboard/sections/:section_id` and set it as the `basename`. For example, if the current location is `/global/fa/teacher_dashboard/sections/28/progress`, the `basename` will be `/global/fa/teacher_dashboard/sections/28`.

Comment (artem-vavilov) [reply apps/src/sites/studio/pages/teacher_dashboard/show.js:92]: https://github.com/code-dot-org/code-dot-org/blob/707e9a12e39fc2c74bbfa2ba8e9184f088f329c8/apps/src/sites/studio/pages/teacher_dashboard/show.js#L86-L92

Comment (wilkie) [reply apps/src/sites/studio/pages/teacher_dashboard/show.js:92]: oh dang I see. Great!

**Review by wilkie — APPROVED** [review]
Tested locally. It looks good.

**Pull request merged**